src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -0,0 +1,575 @@
+    body = {tile_scheduler->vector_region, tile_scheduler->scalar_region};
+    jcp = *reinterpret_cast<const jit_snippets_compile_args*>(tile_scheduler->compile_params);
+}
+void TileSchedulerEmitter::emit_code(const std::vector<size_t> &in,
+                                     const std::vector<size_t> &out,
+                                     const std::vector<size_t> &pool,
+                                     const std::vector<size_t> &gpr) const {
+    validate_arguments(in, out, pool, gpr);
+    emit_impl(in, out, pool, gpr, nullptr);
+}
+void TileSchedulerEmitter::validate_arguments(const std::vector<size_t> &in,
+                                     const std::vector<size_t> &out,
+                                     const std::vector<size_t> &pool,
+                                     const std::vector<size_t> &gpr) const {
+    if (in.size() != 5)
+        IE_THROW() << "TileSchedulerEmitter got invalid number of inputs. Expected 5, got " << in.size();
+    if (out.size() != in[0] + in[1])
+        IE_THROW() << "TileSchedulerEmitter got invalid number of outputs. Expected " << in[0] + in[1] << " , got " << out.size();
+    if (body.size() != 2)
+        IE_THROW() << "TileSchedulerEmitter got invalid body size, expected 2 (vector & scalar TileEmitter), got " << body.size();
+    if (!(std::dynamic_pointer_cast<TileEmitter>(body[0].first) && std::dynamic_pointer_cast<TileEmitter>(body[1].first)))
+        IE_THROW() << "TileSchedulerEmitter can contain only TileEmitters inside its body";
+}
+
+void TileSchedulerEmitter::emit_impl(const std::vector<size_t>& in,
+                                     const std::vector<size_t>& out,
+                                     const std::vector<size_t>& vec_pool,
+                                     const std::vector<size_t>& gpr_pool,
+                                     const ov::intel_cpu::emitter_context *emit_context) const {
+    const size_t num_inputs = in[0];
+    const size_t num_outputs = in[1];
+    const size_t vector_size = in[2];
+    const size_t num_params = num_inputs + num_outputs;
+    const size_t outer_work_amount = jcp.scheduler_dims[0];
+    const size_t inner_work_amount = jcp.scheduler_dims[1];
+    const int64_t harness_num_dims = jcp.output_dims.size() - 1;
+
+    const auto& vector_tile = body[0];
+    const auto& scalar_tile = body[1];
+    const auto& vector_tile_body = std::dynamic_pointer_cast<TileEmitter>(vector_tile.first)->get_nested_code();
+    const auto& scalar_tile_body = std::dynamic_pointer_cast<TileEmitter>(scalar_tile.first)->get_nested_code();
+
+    // remove data ptr regs from the pool, since they should be preserved
+    const auto& data_ptr_reg_idxs(out);
+
+    // It is critical that reg_outer_amount and reg_inner_amount represent the
+    // first two runtime arguments, since they are used to calculating offsets
+    Reg64 reg_outer_amount = Reg64(static_cast<int>(in[3]));
+    Reg64 reg_inner_amount = Reg64(static_cast<int>(in[4]));
```

Comment:
To be honest, I'm confused here because kernel call in cpu plugin uses parameters: `indexes` and `call_args`. But here use call this arguments as `reg_outer_amount` and `reg_inner_amount` and only after that on 203 and 204 lines you use `reg_indexes` and `reg_const_params`. But I think that first of all `in[3]` and `in[4]` are `reg_indexes` and `reg_const_params`.
So could you change these names please?